src/TorchSharp/NN/Linear.cs

```diff
@@ -13,48 +12,53 @@ namespace Modules
     {
         public sealed class Linear : torch.nn.Module<Tensor, Tensor>
         {
-            internal Linear(IntPtr handle, IntPtr boxedHandle) : base(handle, boxedHandle)
+            internal Linear(long inputSize, long outputSize, bool hasBias = true, Device device = null, ScalarType? dtype = null) : base(nameof(Linear))
             {
+                weight = torch.empty(outputSize, inputSize, device: device, dtype: dtype).AsParameter();
+                init.kaiming_uniform_(weight, a: Math.Sqrt(5));
+                
+                if (hasBias) {
+                    bias = torch.empty(outputSize, device: device, dtype: dtype).AsParameter();
+                    var (fanIn, _) = init.CalculateFanInAndFanOut(weight);
+                    var bound = fanIn > 0 ? 1 / Math.Sqrt(fanIn) : 0;
+                    init.uniform_(bias, -bound, bound);
+                }
+                //NOTE: it's important not to call 'RegisterComponents' here.
             }
 
-            public new static Linear Load(string modelPath)
+            public override Tensor forward(Tensor tensor)
             {
-                var res = Module<Tensor, Tensor>.Load(modelPath);
-                return new Linear(res.handle.DangerousGetHandle(), IntPtr.Zero);
+                return torch.nn.functional.linear(tensor, weight, bias);
             }
 
-            public override Tensor forward(Tensor tensor)
+            protected override void Dispose(bool disposing)
             {
-                var res = THSNN_Linear_forward(handle, tensor.Handle);
-                if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                return new Tensor(res);
+                if (disposing) {
+                    weight.Dispose();
+                    if (bias is not null) bias.Dispose();
+                }
             }
 
-            public Parameter? bias {
-                get {
-                    var res = THSNN_Linear_bias(handle);
-                    if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                    return ((res == IntPtr.Zero) ? null : new Parameter(res));
-                }
+            public Parameter bias {
                 set {
-                    THSNN_Linear_set_bias(handle, value?.Handle ?? IntPtr.Zero);
-                    torch.CheckForErrors();
-                    ConditionallyRegisterParameter("bias", value);
+                    _bias = value;
+                    ConditionallyRegisterParameter("bias", bias);
```

Comment:
Will this properly re-register parameter if I manually replace it?

Reply:
Yes, but I found some complications with DisposeScopes that aren't handled. If you overwrite, the bias (and weight, of course) can (and should) be immediately disposed.

Reply:
Might be worth putting that in the `DisposeScope` related section of documentation.